team-aglr/edemame
# React and Express App

This pull request includes the React and Express app. Here's how you can test it:

### **Testing the PR**
- Run `edamame dashboard --start` to start both the React Dashboard and Grafana.
- Go to localhost `3001` and experiment with the dashboard: Start a test, stop a test while its running, delete a prior test, click on "See on AWS" to see if it takes you to your cluster etc.
- If everything is working, try to run a test using the CLI as well. Once the test is finished, the result should appear in your dashboard.
- Once you're done testing, run `edamame dashboard --stop`. This should stop the react and express app, but should leave grafana up and running
- Next  run `edamame grafana --stop` to close grafana

### **Important Note (Grafana)**
- The tests in grafana won't display properly. The buttons in the React dashboard are expecting the filter in Grafana to be based on `name`, but the filters are currently still based on `testId`. Rachel's upcoming commit should fix this, so don't worry if the results aren't appearing properly in grafana yet.

### **Developer Mode:**
- If you want to make change to the React and Express app, you can go into the `backend-server` folder and run `npm run start` to start the express app, and then go to the `frontend` folder and run `npm run start` to run the React app on `localhost:3006` (Note this is different than the production version where the React app is served on `localhost:3001`
- I don't have `nodemon` implemented, so if you make a change in the backend, make sure to restart the server otherwise you won't see the change. Do not add nodemon — it will break a couple of things.
